volatility3/framework/plugins/windows/amcache.py

```diff
@@ -0,0 +1,603 @@
+        # Since we're calling the plugin, make sure we have the plugin's requirements
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Windows kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+            requirements.PluginRequirement(
+                name="hivelist", plugin=hivelist.HiveList, version=(1, 0, 0)
+            ),
+        ]
+
+    @classmethod
+    def get_amcache_hive(
+        cls,
+        context: interfaces.context.ContextInterface,
+        config_path: str,
+        kernel: interfaces.context.ModuleInterface,
+    ) -> Optional[registry.RegistryHive]:
+        """Retrieves the `Amcache.hve` registry hive from the kernel module, if it can be located."""
+        return next(
+            (
+                hive
+                for hive in hivelist.HiveList.list_hives(
+                    context=context,
+                    base_config_path=interfaces.configuration.path_join(
+                        config_path, "hivelist"
+                    ),
+                    layer_name=kernel.layer_name,
+                    symbol_table=kernel.symbol_table_name,
+                    filter_string="amcache",
+                )
+            ),
+            None,
+        )
+
+    @classmethod
+    def parse_file_key(
+        cls, file_key: reg_extensions.CM_KEY_NODE
+    ) -> Iterator[Tuple[str, AmcacheEntry]]:
+        """Parses File entries from the Windows 8 `Root\\File` key.
+
+        :param programs_key: The `Root\\File` registry key.
+
+        :return: An iterator of tuples, where the first member is the program ID string for
+        correlating `Root\\Program` entries, and the second member is the `AmcacheEntry`.
+        """
+        for file_entry_key in file_key.get_subkeys():
+            values = list(file_entry_key.get_values())
+
+            (
+                program_id,
+                path,
+                company,
+                product,
+                sha1_hash,
+                last_mod_raw,
+                last_mod_2_raw,
+                create_time_raw,
+            ) = (
+                _try_get_value(values, "100", reg_extensions.RegValueTypes.REG_SZ),
+                _try_get_value(values, "15", reg_extensions.RegValueTypes.REG_SZ),
+                _try_get_value(values, "1", reg_extensions.RegValueTypes.REG_SZ),
+                _try_get_value(values, "0", reg_extensions.RegValueTypes.REG_SZ),
+                _try_get_value(values, "101", reg_extensions.RegValueTypes.REG_SZ),
+                _try_get_value(values, "11", reg_extensions.RegValueTypes.REG_SZ),
+                _try_get_value(values, "17", reg_extensions.RegValueTypes.REG_SZ),
+                _try_get_value(values, "12", reg_extensions.RegValueTypes.REG_SZ),
+            )
+
+            create_time = (
+                conversion.wintime_to_datetime(create_time_raw)
```

Comment:
I feel like we should now have type hints that do this (eg, HexOrAbsent), but reasonably ok with this.  Just wondering if there's someway to make them more compact or shrink the duplication, but I'm not convinced there is save for a full subfunction, so this is probably fine.